File: pkg/models/campaign.go
package models

import (
	"encoding/json"
	"time"

	"github.com/google/uuid"
)

type CampaignStatus string

const (
	CampaignStatusActive    CampaignStatus = "active"
	CampaignStatusPaused    CampaignStatus = "paused"
	CampaignStatusCompleted CampaignStatus = "completed"
	CampaignStatusArchived  CampaignStatus = "archived"
)

type Campaign struct {
	ID            uuid.UUID       `json:"id"`
	Name          string          `json:"name"`
	Description   *string         `json:"description,omitempty"`
	Status        CampaignStatus  `json:"status"`
	InputType     *string         `json:"input_type,omitempty"`
	Keywords      []string        `json:"keywords,omitempty"`
	SeedProfiles  []string        `json:"seed_profiles,omitempty"`
	AIQueries     []string        `json:"ai_queries,omitempty"`
	FollowerDepth int             `json:"follower_depth"`
	MaxResults    *int            `json:"max_results,omitempty"`
	UserID        *uuid.UUID      `json:"user_id,omitempty"`
	BudgetCents   int64           `json:"budget_cents"`
	SpendCents    int64           `json:"spend_cents"`
	CreatedAt     time.Time       `json:"created_at"`
	UpdatedAt     time.Time       `json:"updated_at"`
	Metadata      json.RawMessage `json:"metadata,omitempty"`
}

File: services/frontend/types/campaign.ts
export type CampaignStatus = 'active' | 'paused' | 'completed' | 'archived'

export interface Campaign {
  id: string
  name: string
  description?: string
  status: CampaignStatus
  input_type?: string
  keywords: string[]
  seed_profiles: string[]
  ai_queries: string[]
  follower_depth: number
  max_results?: number
  user_id?: string
  budget_cents: number
  spend_cents: number
  created_at: string
  updated_at: string
  metadata?: Record<string, unknown>
}

export interface CreateCampaignInput {
  name: string
  description?: string
  input_type?: string
  keywords?: string[]
  seed_profiles?: string[]
  ai_queries?: string[]
  follower_depth?: number
  max_results?: number
  budget_cents?: number
}

export interface UpdateCampaignInput {
  name?: string
  description?: string
  status?: CampaignStatus
  max_results?: number
  budget_cents?: number
}
